rust/src/lib/nm/query_apply/device.rs

```diff
@@ -1,43 +1,39 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    nm::nm_dbus::NmDevice,
-    nm::settings::{
-        NM_SETTING_BOND_SETTING_NAME, NM_SETTING_BRIDGE_SETTING_NAME,
-        NM_SETTING_DUMMY_SETTING_NAME, NM_SETTING_HSR_SETTING_NAME,
-        NM_SETTING_INFINIBAND_SETTING_NAME, NM_SETTING_LOOPBACK_SETTING_NAME,
-        NM_SETTING_MACSEC_SETTING_NAME, NM_SETTING_MACVLAN_SETTING_NAME,
-        NM_SETTING_OVS_BRIDGE_SETTING_NAME, NM_SETTING_OVS_IFACE_SETTING_NAME,
-        NM_SETTING_VETH_SETTING_NAME, NM_SETTING_VLAN_SETTING_NAME,
-        NM_SETTING_VRF_SETTING_NAME, NM_SETTING_VXLAN_SETTING_NAME,
-        NM_SETTING_WIRED_SETTING_NAME,
-    },
+    nm::nm_dbus::{NmDevice, NmIfaceType},
     InterfaceType,
 };
 
+fn nm_iface_type_to_nmstate(nm_iface_type: &NmIfaceType) -> InterfaceType {
+    match nm_iface_type {
+        NmIfaceType::Bridge => InterfaceType::LinuxBridge,
+        NmIfaceType::Bond => InterfaceType::Bond,
+        NmIfaceType::Ethernet => InterfaceType::Ethernet,
+        NmIfaceType::OvsBridge => InterfaceType::OvsBridge,
+        NmIfaceType::OvsIface => InterfaceType::OvsInterface,
+        NmIfaceType::Vlan => InterfaceType::Vlan,
+        NmIfaceType::Vxlan => InterfaceType::Vxlan,
+        NmIfaceType::Dummy => InterfaceType::Dummy,
+        NmIfaceType::Macvlan => InterfaceType::MacVlan,
+        NmIfaceType::Vrf => InterfaceType::Vrf,
+        // We unify the Veth to into InterfaceType::Ethernet to simplify
+        // work on using veth as plain ethernet interface
+        NmIfaceType::Veth => InterfaceType::Ethernet,
+        NmIfaceType::Infiniband => InterfaceType::InfiniBand,
+        NmIfaceType::Loopback => InterfaceType::Loopback,
+        NmIfaceType::Macsec => InterfaceType::MacSec,
+        NmIfaceType::Hsr => InterfaceType::Hsr,
+        _ => InterfaceType::Other(nm_iface_type.to_string()),
+    }
+}
```

Comment:
Same, this function would be better placed at nm_dbus/connection/iface_type.rs

Reply:
The `nm_dbus` cannot depend on `nmstate::InterfaceType`. THe `nm_dbus` folder was supposed to be standalone and could be released as a standalone crate in the future.